packages/database_v2/migrations/default/1632668838948_uri-hash/up.sql

```diff
@@ -0,0 +1,103 @@
+
+BEGIN TRANSACTION;
+
+--- NFT Asset
+
+-- Create a derived column
+ALTER TABLE nft_asset
+    ADD COLUMN token_uri_hash bytea
+    GENERATED ALWAYS AS (sha256(token_uri::bytea))
+    STORED;
+
+-- Remove primary key constraint
+ALTER TABLE nft_asset
+    DROP CONSTRAINT nft_asset_pkey;
+
+-- Add primary key constraint to new column
+ALTER TABLE nft_asset
+    ADD CONSTRAINT nft_asset_pkey PRIMARY KEY (token_uri_hash);
+
+
+--- NFT
+
+-- Add token_uri_hash field
+ALTER TABLE nft
+    ADD COLUMN token_uri_hash bytea;
+
+-- Populate column with hashes
+UPDATE nft
+    SET token_uri_hash = sha256(token_uri::bytea);
```

Comment:
@ribasushi this seems to fail with

```
sql_error_code = 22P02 ERROR:  invalid input syntax for type bytea
```

Which if this is correct https://stackoverflow.com/questions/36063777/postgresql-9-5-syntax-error is due to the fact that some `token_uri` contains invalid escape which I'd guess is one of those svgs that have `<\image>`.


I clearly don't seem to know the right way to turn text into bytea to hash it, mind educating me on this ?

Reply:
Fml, I never had to do this myself.
Instead of `sometext::BYTEA` you apparently need `CONVERT_TO( sometext, 'SQL_ASCII' )`
If it doesn't like the chosen encoding - go with something making sense in your context ( `LATIN1` or similar )